### ISSUE TITLE
problems with mockRequest in Express sessions

### ISSUE THREAD

**Issue by marneborn** [howardabrams/node-mocks-http#52]
I'm writing some tests for my sessions handling and have found two minor problems.
In express 3, sessions throws an error if req.signedCookies is not set.
```    var unsignedCookie = req.signedCookies[key];```
This seems to have been fixed in express 4, but that's based on reading the code, not trying it...
.
Second, mockRequest.session is set to {} by default but the session isn't created if req.session is truthy
```
  return function session(req, res, next) {
    // self-awareness
    if (req.session) return next();
```
The mockRequest value is set by this:
```    mockRequest.session = (options.session) ? options.session : {};```
So any falsey option gets the default value, so I can't do { session : null }.
As a side note, mockRequest({ body : 0 }) will result in req.body = {}.

Workarounds simply
```
var req = httpMocks.createRequest({});
req.signedCookies = {}
delete req.session
```

**Comment by JohnnyEstilles** [comment]
@marneborn, thanks for reporting this. I will try to reproduce both these issues.

**Comment by JohnnyEstilles** [comment]
I have been able to verify that `req.session` should be `undefined` by default. I will work on a fix for this issue ASAP.

**Comment by JohnnyEstilles** [comment]
I have also been able to verify that `req.signedCookies` should also be `undefined` by default. It is initialized to an empty object by [cookie-parse](https://github.com/expressjs/cookie-parser) when used. Will also work on a fix for this ASAP.

**Comment by JohnnyEstilles** [comment]
@marneborn, I've implemented a fix for your issues (see release notes for [v1.2.7](https://github.com/howardabrams/node-mocks-http/releases/tag/v1.2.7)). The following is now the default behavior or `mockResponse`:

- `.session` defaults to undefined (must explicitly be set by passing options to `.createRequest()` or using `._setSessionVariable()`)
- `.signedCookies` defaults to undefined (now you can set by passing options to `.createRequest()` or using `._setSignedCookiesVariable()`)

Regarding your two specific issues:

> In express 3, sessions throws an error if req.signedCookies is not set.
```js
var unsignedCookie = req.signedCookies[key];
```

... should now throw an error as expected, unless `.signedCookies` has been explicitly set (see explanation above).

>  mockRequest.session is set to {} by default but the session isn't created if req.session is truthy
```js
return function session(req, res, next) {
    // self-awareness
    if (req.session) return next();
```

... should now work as expected, `req.session` will resolve to a `false` value and `return next()` will be executed, unless `.session` has been explicitly set (see explanation above)..

Please let us know if you have any other issues.

Closed via  [v1.2.7](https://github.com/howardabrams/node-mocks-http/releases/tag/v1.2.7).
